### ISSUE TITLE
[HOLD for payment 2022-09-23] [$500] Update the staging App to use the staging API

### ISSUE THREAD

**Issue by flodnv** [Expensify/App#10163]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Problem:** NewDot staging points to the production version of our API. This means that we miss API bugs deployed to staging, they then creep up to production and affect customers.

**Solution:** Point NewDot staging to [staging.expensify.com](http://staging.expensify.com/). Replace the current toggle `Use Secure Staging Server` with `Use Staging Server` (which will use the staging API for both www and secure, and be enabled by default on the staging App) such that mobile apps and desktop can easily switch back and forth.

See [this comment](https://github.com/Expensify/App/issues/10163#issuecomment-1203858142) for more context about the expected solution.

## Action Performed:
Use the staging app (eg staging.new.expensify.com)

## Expected Result:
It uses the staging API (staging.expensify.com)

## Actual Result:
It uses the production API (www.expensify.com)

## Workaround:
None

## Platform:

All of them.

**Internal slack conversation:**  https://expensify.slack.com/archives/CC7NECV4L/p1659081597417989

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by NicMendonca** [comment]
Job post: https://www.upwork.com/jobs/~01483fb4b26ac8eba4

**Comment by wildan-m** [comment]
Hi,

Is it for an external contributor? How can we reproduce the issue locally?

I've tried to use localhost and also using ngrok, but the API URL correctly pointed to the desired URL

<img width="659" alt="image" src="https://user-images.githubusercontent.com/11847279/182550357-6372f38e-9ffe-46ce-9dd7-adcd8acb1632.png">

**Comment by flodnv** [comment]
On staging.new.expensify.com and new.expensify.com, you should see requests going to www.expensify.com. This update is about pointing staging.new.expensify.com to staging.expensify.com

**Comment by wildan-m** [comment]
@flodnv did you mean to refactor the caption and variable to `Use Staging Server` and `shouldUseStaging` then make the `shouldUseStaging` default value to `true`?

```diff
diff --git a/src/libs/HttpUtils.js b/src/libs/HttpUtils.js
index 3e93e3182..13ac4cc3d 100644
--- a/src/libs/HttpUtils.js
+++ b/src/libs/HttpUtils.js
@@ -5,10 +5,10 @@ import CONST from '../CONST';
 import ONYXKEYS from '../ONYXKEYS';
 import HttpsError from './Errors/HttpsError';
 
-let shouldUseSecureStaging = false;
+let shouldUseStaging = false;
 Onyx.connect({
     key: ONYXKEYS.USER,
-    callback: val => shouldUseSecureStaging = (val && _.isBoolean(val.shouldUseSecureStaging)) ? val.shouldUseSecureStaging : false,
+    callback: val => shouldUseStaging = (val && _.isBoolean(val.shouldUseStaging)) ? val.shouldUseStaging : false,
 });
 
 let shouldFailAllRequests = false;
@@ -96,7 +96,7 @@ function xhr(command, data, type = CONST.NETWORK.METHOD.POST, shouldUseSecure =
     });
     let apiRoot = shouldUseSecure ? CONFIG.EXPENSIFY.SECURE_EXPENSIFY_URL : CONFIG.EXPENSIFY.URL_API_ROOT;
 
-    if (shouldUseSecure && shouldUseSecureStaging) {
+    if (shouldUseSecure && shouldUseStaging) {
         apiRoot = CONST.STAGING_SECURE_URL;
     }
 
diff --git a/src/libs/actions/User.js b/src/libs/actions/User.js
index d41fcf3ef..e76374cbd 100644
--- a/src/libs/actions/User.js
+++ b/src/libs/actions/User.js
@@ -426,10 +426,10 @@ function updateChatPriorityMode(mode) {
 }
 
 /**
- * @param {Boolean} shouldUseSecureStaging
+ * @param {Boolean} shouldUseStaging
  */
-function setShouldUseSecureStaging(shouldUseSecureStaging) {
-    Onyx.merge(ONYXKEYS.USER, {shouldUseSecureStaging});
+function setShouldUseStaging(shouldUseStaging) {
+    Onyx.merge(ONYXKEYS.USER, {shouldUseStaging});
 }
 
 function clearUserErrorMessage() {
wildanmuhlis@Wildans-MacBook-Air App % git diff src/components/TestToolMenu.js src/libs/HttpUtils.js src/libs/actions/User.js src/libs/actions/User.js src/pages/settings/PreferencesPage.js
diff --git a/src/components/TestToolMenu.js b/src/components/TestToolMenu.js
index f04ae6a26..9c0860da6 100644
--- a/src/components/TestToolMenu.js
+++ b/src/components/TestToolMenu.js
@@ -18,7 +18,7 @@ const propTypes = {
     /** User object in Onyx */
     user: PropTypes.shape({
         /** Whether we should use the staging version of the secure API server */
-        shouldUseSecureStaging: PropTypes.bool,
+        shouldUseStaging: PropTypes.bool,
     }),
 
     /** Network object in Onyx */
@@ -27,7 +27,7 @@ const propTypes = {
 
 const defaultProps = {
     user: {
-        shouldUseSecureStaging: false,
+        shouldUseStaging: true,
     },
 };
 
@@ -39,10 +39,10 @@ const TestToolMenu = props => (
 
         {/* Option to switch from using the staging secure endpoint or the production secure endpoint.
         This enables QA and internal testers to take advantage of sandbox environments for 3rd party services like Plaid and Onfido. */}
-        <TestToolRow title="Use Secure Staging Server">
+        <TestToolRow title="Use Staging Server">
             <Switch
-                isOn={props.user.shouldUseSecureStaging || false}
-                onToggle={() => User.setShouldUseSecureStaging(!props.user.shouldUseSecureStaging)}
+                isOn={props.user.shouldUseStaging || false}
+                onToggle={() => User.setShouldUseStaging(!props.user.shouldUseStaging)}
             />
         </TestToolRow>
 
diff --git a/src/libs/HttpUtils.js b/src/libs/HttpUtils.js
index 3e93e3182..13ac4cc3d 100644
--- a/src/libs/HttpUtils.js
+++ b/src/libs/HttpUtils.js
@@ -5,10 +5,10 @@ import CONST from '../CONST';
 import ONYXKEYS from '../ONYXKEYS';
 import HttpsError from './Errors/HttpsError';
 
-let shouldUseSecureStaging = false;
+let shouldUseStaging = false;
 Onyx.connect({
     key: ONYXKEYS.USER,
-    callback: val => shouldUseSecureStaging = (val && _.isBoolean(val.shouldUseSecureStaging)) ? val.shouldUseSecureStaging : false,
+    callback: val => shouldUseStaging = (val && _.isBoolean(val.shouldUseStaging)) ? val.shouldUseStaging : false,
 });
 
 let shouldFailAllRequests = false;
@@ -96,7 +96,7 @@ function xhr(command, data, type = CONST.NETWORK.METHOD.POST, shouldUseSecure =
     });
     let apiRoot = shouldUseSecure ? CONFIG.EXPENSIFY.SECURE_EXPENSIFY_URL : CONFIG.EXPENSIFY.URL_API_ROOT;
 
-    if (shouldUseSecure && shouldUseSecureStaging) {
+    if (shouldUseSecure && shouldUseStaging) {
         apiRoot = CONST.STAGING_SECURE_URL;
     }
 
diff --git a/src/libs/actions/User.js b/src/libs/actions/User.js
index d41fcf3ef..e76374cbd 100644
--- a/src/libs/actions/User.js
+++ b/src/libs/actions/User.js
@@ -426,10 +426,10 @@ function updateChatPriorityMode(mode) {
 }
 
 /**
- * @param {Boolean} shouldUseSecureStaging
+ * @param {Boolean} shouldUseStaging
  */
-function setShouldUseSecureStaging(shouldUseSecureStaging) {
-    Onyx.merge(ONYXKEYS.USER, {shouldUseSecureStaging});
+function setShouldUseStaging(shouldUseStaging) {
+    Onyx.merge(ONYXKEYS.USER, {shouldUseStaging});
 }
 
 function clearUserErrorMessage() {
@@ -484,7 +484,7 @@ export {
     isBlockedFromConcierge,
     subscribeToUserEvents,
     updatePreferredSkinTone,
-    setShouldUseSecureStaging,
+    setShouldUseStaging,
     clearUserErrorMessage,
     subscribeToExpensifyCardUpdates,
     updateFrequentlyUsedEmojis,
diff --git a/src/pages/settings/PreferencesPage.js b/src/pages/settings/PreferencesPage.js
index c96edfffd..a6f123beb 100755
--- a/src/pages/settings/PreferencesPage.js
+++ b/src/pages/settings/PreferencesPage.js
@@ -30,7 +30,7 @@ const propTypes = {
     user: PropTypes.shape({
         /** Whether or not the user is subscribed to news updates */
         isSubscribedToNewsletter: PropTypes.bool,
-        shouldUseSecureStaging: PropTypes.bool,
+        shouldUseStaging: PropTypes.bool,
     }),
 
     ...withLocalizePropTypes,
```

**Comment by mountiny** [comment]
@wildan-m I think there is a bit of confusion. We have two API servers, one classic and one secure. Both of them have production and staging versions.

What the toggle does currently is that it will point any Secure API calls to staging. However, the classic API calls are all pointed at the production API endpoint, which gives us no option to test our API in staging with this App. 

What we want is to refactor this toggle to not only switch the Secure APIs but also the classic ones. It can be done externally but context is missing, I will try to add it now.

You would have to add 2 new API endpoints here https://github.com/Expensify/App/blob/41c7054db7758342394d07617a76f64bfbeef8a5/src/CONFIG.js#L7-L23

The classic API staging url is:
```
const stagingExpensifyURL = Url.addTrailingForwardSlash(lodashGet(Config, 'EXPENSIFY_URL', 'https://staging.expensify.com/'));
```

And the secure is:
```
const stagingSecureExpensifyUrl = Url.addTrailingForwardSlash(lodashGet(
    Config, 'SECURE_EXPENSIFY_URL', 'https://staging-secure.expensify.com/',
));
```
That one is already defined here: https://github.com/Expensify/App/blob/41c7054db7758342394d07617a76f64bfbeef8a5/src/CONST.js#L230

However, I think we should move it to the Config.js file with the other links.

Then the toggle should not only redirect the secure.expensify calls abut also any expensify.com/api.php calls to staging as defined above.

Does this make sense?

**Comment by danieldoglas** [comment]
Closed the [issue](https://github.com/Expensify/App/issues/10084) that created this discussion.

**Comment by mountiny** [comment]
@NicMendonca We might want to double the price on this one! Thank you 🙇

**Comment by NicMendonca** [comment]
Price doubled: https://www.upwork.com/jobs/~01483fb4b26ac8eba4

**Comment by mdneyazahmad** [comment]
## Proposal

We need to do following changes.
1. Add `IS_IN_STAGING: ENVIRONMENT === CONST.ENVIRONMENT.STAGING,` in `CONFIG.js`.
2. Add `STAGING_URL: 'https://staging.expensify.com/',` in `CONST.js`.

3. Change `title` and `isOn` condition.
https://github.com/Expensify/App/blob/22e39f9de1c1fbace6e7a0e4c06f364310129895/src/components/TestToolMenu.js#L42-L47

```javascript
// this will be on by default (we do not show this tools in production)
isOn={lodashGet(props, 'user.shouldUseSecureStaging', true)}
```

4. Update the condition in `HttpUtils.js`.

https://github.com/Expensify/App/blob/a70aa0cd498820662bd9ac4138026095e3e0ed31/src/libs/HttpUtils.js#L8-L12

```diff
-    callback: val => shouldUseSecureStaging = (val && _.isBoolean(val.shouldUseSecureStaging)) ? val.shouldUseSecureStaging : false,
+    callback: val => shouldUseSecureStaging = lodashGet(val, 'shouldUseSecureStaging', true),
```

4. Update the condition in to use api based on environment`HttpUtils.js`.

https://github.com/Expensify/App/blob/a70aa0cd498820662bd9ac4138026095e3e0ed31/src/libs/HttpUtils.js#L97-L101

```diff
-    let apiRoot = shouldUseSecure ? CONFIG.EXPENSIFY.SECURE_EXPENSIFY_URL : CONFIG.EXPENSIFY.URL_API_ROOT;
 
-    if (shouldUseSecure && shouldUseSecureStaging) {
-          apiRoot = CONST.STAGING_SECURE_URL;
-    }
+    let apiRoot = CONFIG.IS_IN_STAGING ? CONST.STAGING_URL : CONFIG.EXPENSIFY.URL_API_ROOT;
+
+    if (shouldUseSecure) {
+        apiRoot = CONFIG.EXPENSIFY.SECURE_EXPENSIFY_URL;
+    }
+
+    if (shouldUseSecure && CONFIG.IS_IN_STAGING && shouldUseSecureStaging) {
+        apiRoot = CONST.STAGING_SECURE_URL;
+    }
```

commit: https://github.com/mdneyazahmad/App/commit/af713fbf9412be5c1d753ebc5b2a353a3f4ea71f

Finally, we can do some refactor like changing names, placing some code in different file etc.

**Comment by mdneyazahmad** [comment]
Sorry, I see condition on 4 is not correct I will update it soon but you got the idea.

**Comment by Santhosh-Sellavel** [comment]
> Finally, we can do some refactor like changing names, placing some code in different file etc.

@mdneyazahmad Can you update your proposal to meet all the changes mentioned [here](https://github.com/Expensify/App/issues/10163#issuecomment-1203858142)

**Comment by Santhosh-Sellavel** [comment]
The proposal doesn't meet the requirements @mdneyazahmad 

Can you check this https://github.com/Expensify/App/issues/10163#issuecomment-1203858142 and update proposal thanks!
cc: @mountiny

**Comment by mountiny** [comment]
@mdneyazahmad We need to update also the other API endpoint, there are two. Secure which is already working with the switch and the other one which is not an dI have described it in the comment above.

**Comment by wildan-m** [comment]
### Proposal

1. Add Two new endpoints in `src/CONFIG.js `
```diff
diff --git a/src/CONFIG.js b/src/CONFIG.js
index a60b17e1e..6119e0f81 100644
--- a/src/CONFIG.js
+++ b/src/CONFIG.js
@@ -14,6 +14,10 @@ const secureNgrokURL = Url.addTrailingForwardSlash(lodashGet(Config, 'SECURE_NGR
 const secureExpensifyUrl = Url.addTrailingForwardSlash(lodashGet(
     Config, 'SECURE_EXPENSIFY_URL', 'https://secure.expensify.com/',
 ));
+const stagingExpensifyURL = Url.addTrailingForwardSlash(lodashGet(Config, 'STAGING_EXPENSIFY_URL', 'https://staging.expensify.com/'));
+const stagingSecureExpensifyUrl = Url.addTrailingForwardSlash(lodashGet(
+    Config, 'STAGING_SECURE_EXPENSIFY_URL', 'https://staging-secure.expensify.com/',
+));
 const useNgrok = lodashGet(Config, 'USE_NGROK', 'false') === 'true';
 const useWebProxy = lodashGet(Config, 'USE_WEB_PROXY', 'true') === 'true';
 const expensifyComWithProxy = getPlatform() === 'web' && useWebProxy ? '/' : expensifyURL;
@@ -50,6 +54,8 @@ export default {
         PARTNER_PASSWORD: lodashGet(Config, 'EXPENSIFY_PARTNER_PASSWORD', 'e21965746fd75f82bb66'),
         EXPENSIFY_CASH_REFERER: 'ecash',
         CONCIERGE_URL: conciergeUrl,
+        STAGING_EXPENSIFY_URL: stagingExpensifyURL,
+        STAGING_SECURE_EXPENSIFY_URL: stagingSecureExpensifyUrl,
     },
     IS_IN_PRODUCTION: Platform.OS === 'web' ? process.env.NODE_ENV === 'production' : !__DEV__,
     IS_USING_LOCAL_WEB: useNgrok || expensifyURLRoot.includes('dev'),
```

2. Modify toggle logic

```diff
diff --git a/src/libs/HttpUtils.js b/src/libs/HttpUtils.js
index 3e93e3182..666fdf4c2 100644
--- a/src/libs/HttpUtils.js
+++ b/src/libs/HttpUtils.js
@@ -5,10 +5,10 @@ import CONST from '../CONST';
 import ONYXKEYS from '../ONYXKEYS';
 import HttpsError from './Errors/HttpsError';
 
-let shouldUseSecureStaging = false;
+let shouldUseStaging = true;
 Onyx.connect({
     key: ONYXKEYS.USER,
-    callback: val => shouldUseSecureStaging = (val && _.isBoolean(val.shouldUseSecureStaging)) ? val.shouldUseSecureStaging : false,
+    callback: val => shouldUseStaging = (val && _.isBoolean(val.shouldUseStaging)) ? val.shouldUseStaging : false,
 });
 
 let shouldFailAllRequests = false;
@@ -94,10 +94,17 @@ function xhr(command, data, type = CONST.NETWORK.METHOD.POST, shouldUseSecure =
 
         formData.append(key, val);
     });
-    let apiRoot = shouldUseSecure ? CONFIG.EXPENSIFY.SECURE_EXPENSIFY_URL : CONFIG.EXPENSIFY.URL_API_ROOT;
+    let apiRoot = CONFIG.EXPENSIFY.URL_API_ROOT;
 
-    if (shouldUseSecure && shouldUseSecureStaging) {
-        apiRoot = CONST.STAGING_SECURE_URL;
+    if (shouldUseStaging) {
+        apiRoot = CONFIG.EXPENSIFY.STAGING_EXPENSIFY_URL;
+    }
+
+    if (shouldUseSecure) {
+        apiRoot = CONFIG.EXPENSIFY.SECURE_EXPENSIFY_URL;
+        if (shouldUseStaging) {
+            apiRoot = CONFIG.EXPENSIFY.STAGING_SECURE_EXPENSIFY_URL;
+        }
     }
 
     return processHTTPRequest(`${apiRoot}api?command=${command}`, type, formData, data.canCancel);
```

3. Move `STAGING_SECURE_URL` to /src/CONFIG.js

```diff
diff --git a/src/CONST.js b/src/CONST.js
index 7ef3640fb..fd80bad03 100755
--- a/src/CONST.js
+++ b/src/CONST.js
@@ -232,7 +232,6 @@ const CONST = {
     MANAGE_CARDS_URL: 'domain_companycards',
     FEES_URL: `${USE_EXPENSIFY_URL}/fees`,
     CFPB_PREPAID_URL: 'https://cfpb.gov/prepaid',
-    STAGING_SECURE_URL: 'https://staging-secure.expensify.com/',
     STAGING_NEW_EXPENSIFY_URL: 'https://staging.new.expensify.com',
 
     // Use Environment.getEnvironmentURL to get the complete URL with port number
```
4. Modify example

```diff
diff --git a/.env.example b/.env.example
index cf13ef583..01ec169ad 100644
--- a/.env.example
+++ b/.env.example
@@ -1,6 +1,8 @@
 NEW_EXPENSIFY_URL=https://new.expensify.com/
 SECURE_EXPENSIFY_URL=https://secure.expensify.com.dev/
 EXPENSIFY_URL=https://www.expensify.com.dev/
+STAGING_SECURE_EXPENSIFY_URL=https://staging-secure.expensify.com/
+STAGING_EXPENSIFY_URL=https://staging.expensify.com/
 EXPENSIFY_PARTNER_NAME=chat-expensify-com
 EXPENSIFY_PARTNER_PASSWORD=e21965746fd75f82bb66
 PUSHER_APP_KEY=ac6d22b891daae55283a
```

5. Name and caption refactor

```diff
diff --git a/src/CONFIG.js b/src/CONFIG.js
index a60b17e1e..6119e0f81 100644
--- a/src/CONFIG.js
+++ b/src/CONFIG.js
@@ -14,6 +14,10 @@ const secureNgrokURL = Url.addTrailingForwardSlash(lodashGet(Config, 'SECURE_NGR
 const secureExpensifyUrl = Url.addTrailingForwardSlash(lodashGet(
     Config, 'SECURE_EXPENSIFY_URL', 'https://secure.expensify.com/',
 ));
+const stagingExpensifyURL = Url.addTrailingForwardSlash(lodashGet(Config, 'STAGING_EXPENSIFY_URL', 'https://staging.expensify.com/'));
+const stagingSecureExpensifyUrl = Url.addTrailingForwardSlash(lodashGet(
+    Config, 'STAGING_SECURE_EXPENSIFY_URL', 'https://staging-secure.expensify.com/',
+));
 const useNgrok = lodashGet(Config, 'USE_NGROK', 'false') === 'true';
 const useWebProxy = lodashGet(Config, 'USE_WEB_PROXY', 'true') === 'true';
 const expensifyComWithProxy = getPlatform() === 'web' && useWebProxy ? '/' : expensifyURL;
@@ -50,6 +54,8 @@ export default {
         PARTNER_PASSWORD: lodashGet(Config, 'EXPENSIFY_PARTNER_PASSWORD', 'e21965746fd75f82bb66'),
         EXPENSIFY_CASH_REFERER: 'ecash',
         CONCIERGE_URL: conciergeUrl,
+        STAGING_EXPENSIFY_URL: stagingExpensifyURL,
+        STAGING_SECURE_EXPENSIFY_URL: stagingSecureExpensifyUrl,
     },
     IS_IN_PRODUCTION: Platform.OS === 'web' ? process.env.NODE_ENV === 'production' : !__DEV__,
     IS_USING_LOCAL_WEB: useNgrok || expensifyURLRoot.includes('dev'),

diff --git a/src/components/TestToolMenu.js b/src/components/TestToolMenu.js
index f04ae6a26..9c0860da6 100644
--- a/src/components/TestToolMenu.js
+++ b/src/components/TestToolMenu.js
@@ -18,7 +18,7 @@ const propTypes = {
     /** User object in Onyx */
     user: PropTypes.shape({
         /** Whether we should use the staging version of the secure API server */
-        shouldUseSecureStaging: PropTypes.bool,
+        shouldUseStaging: PropTypes.bool,
     }),
 
     /** Network object in Onyx */
@@ -27,7 +27,7 @@ const propTypes = {
 
 const defaultProps = {
     user: {
-        shouldUseSecureStaging: false,
+        shouldUseStaging: true,
     },
 };
 
@@ -39,10 +39,10 @@ const TestToolMenu = props => (
 
         {/* Option to switch from using the staging secure endpoint or the production secure endpoint.
         This enables QA and internal testers to take advantage of sandbox environments for 3rd party services like Plaid and Onfido. */}
-        <TestToolRow title="Use Secure Staging Server">
+        <TestToolRow title="Use Staging Server">
             <Switch
-                isOn={props.user.shouldUseSecureStaging || false}
-                onToggle={() => User.setShouldUseSecureStaging(!props.user.shouldUseSecureStaging)}
+                isOn={props.user.shouldUseStaging || false}
+                onToggle={() => User.setShouldUseStaging(!props.user.shouldUseStaging)}
             />
         </TestToolRow>
 

diff --git a/src/libs/actions/User.js b/src/libs/actions/User.js
index a7b71d9fd..3f06a5536 100644
--- a/src/libs/actions/User.js
+++ b/src/libs/actions/User.js
@@ -426,10 +426,10 @@ function updateChatPriorityMode(mode) {
 }
 
 /**
- * @param {Boolean} shouldUseSecureStaging
+ * @param {Boolean} shouldUseStaging
  */
-function setShouldUseSecureStaging(shouldUseSecureStaging) {
-    Onyx.merge(ONYXKEYS.USER, {shouldUseSecureStaging});
+function setShouldUseStaging(shouldUseStaging) {
+    Onyx.merge(ONYXKEYS.USER, {shouldUseStaging});
 }
 
 function clearUserErrorMessage() {
@@ -484,7 +484,7 @@ export {
     isBlockedFromConcierge,
     subscribeToUserEvents,
     updatePreferredSkinTone,
-    setShouldUseSecureStaging,
+    setShouldUseStaging,
     clearUserErrorMessage,
     subscribeToExpensifyCardUpdates,
     updateFrequentlyUsedEmojis,
diff --git a/src/pages/settings/PreferencesPage.js b/src/pages/settings/PreferencesPage.js
index c96edfffd..a6f123beb 100755
--- a/src/pages/settings/PreferencesPage.js
+++ b/src/pages/settings/PreferencesPage.js
@@ -30,7 +30,7 @@ const propTypes = {
     user: PropTypes.shape({
         /** Whether or not the user is subscribed to news updates */
         isSubscribedToNewsletter: PropTypes.bool,
-        shouldUseSecureStaging: PropTypes.bool,
+        shouldUseStaging: PropTypes.bool,
     }),
 
     ...withLocalizePropTypes,
```

**Comment by mountiny** [comment]
@Santhosh-Sellavel Thought on the updated proposal from @mdneyazahmad and new proposal from @wildan-m?

**Comment by NicMendonca** [comment]
@Santhosh-Sellavel bump!

**Comment by Santhosh-Sellavel** [comment]
@mountiny 
@mdneyazahmad's [proposal](https://github.com/Expensify/App/issues/10163#issuecomment-1222393896) covers all things, except the URLs are not defined in CONFIG.js

@wildan-m's [proposal](https://github.com/Expensify/App/issues/10163#issuecomment-1222393896) corrected their mistake from their old proposal.

I'll let you choose between the two proposals. I'll favor @mdneyazahmad as theirs seems to be the first complete proposal here. 

C+ Reviewed
🎀 👀 🎀

**Comment by mountiny** [comment]
I will assign @mdneyazahmad since they had a complete proposal earlier. I am sorry @wildan-m for this, I am sure there is many more issues your help will be much appreciated.

Thank you @Santhosh-Sellavel

**Comment by wildan-m** [comment]
> I will assign @mdneyazahmad since they had a complete proposal earlier. I am sorry @wildan-m for this, I am sure there is many more issues your help will be much appreciated.
> 
> Thank you @Santhosh-Sellavel 

no worries, thanks for your feedback

**Comment by mountiny** [comment]
@mdneyazahmad @Santhosh-Sellavel Lets try to get this out sooner than later if possible. This is very handy feature for our QA to catch some API mistakes in staging before it hits production 🙇

**Comment by Santhosh-Sellavel** [comment]
@mountiny bump!

**Comment by mountiny** [comment]
I think you can make QA steps where you instruct the tester to go to the Network tabs and to observe what API the event is sent to (if it is staging or not).

The PR has not been made ready for review yet, if you do so, I will be assigned to review together with Santosh

**Comment by mountiny** [comment]
I will revert the PR

**Comment by Santhosh-Sellavel** [comment]
@mountiny I believe the reason for the regression is beyond our control, as this can be found, only testing after deploying the build.

@mountiny What's next here?

**Comment by mountiny** [comment]
@Santhosh-Sellavel We are adding an update to our servers which should hopefully resolve this, but just decided to revert this before weekend to make sure. 

Any ideas how we could test if the requests work when made from `staging.new.expensify.com` before we would merge this PR again? Worst case we can just try to merge this and deploy once the servers will be updated (we havent realized this before this change)

**Comment by mountiny** [comment]
Thats what we are adding! 🎉 should be good then. Thanks! Gonna ask @mdneyazahmad to raise the PR again once the server is ready

**Comment by mountiny** [comment]
@mdneyazahmad The changes were deployed in the server, can you please raise the PR again? You can find the `staging.expensify.com` server in the curl output.

@NicMendonca Even though there was a regression from this PR, I would say we should not treat this agains the contributors or c+. The regression was caused by a servers settings, which we did not realize before.

**Comment by mountiny** [comment]
Thanks!

**Comment by mountiny** [comment]
Removing the hold for payment now and we will wait for the second PR to hit production before the payment would be paid out to ensure the regression period is met.

**Comment by NicMendonca** [comment]
@mdneyazahmad don't forget to apply for [the job](https://www.upwork.com/jobs/~01483fb4b26ac8eba4) in Upwork please!

@Santhosh-Sellavel don't forget to accept the invite to the job :)

**Comment by Santhosh-Sellavel** [comment]
I am getting an error while accepting
@NicMendonca Can you cancel & resent please, thanks.

**Comment by NicMendonca** [comment]
@Santhosh-Sellavel can you try again please?

@mdneyazahmad you'll just need to accept the offer. 

Thanks both!

**Comment by NicMendonca** [comment]
@mdneyazahmad paid 🎉

**Comment by Santhosh-Sellavel** [comment]
Done thanks @NicMendonca!

**Comment by NicMendonca** [comment]
fab, everyone is paid! Happy weekend! 😊

**Comment by iwiznia** [comment]
Hello, I just realized this is not working for Android (I assume neither for iOS, but did not test). I have the staging app (currently 1.2.10) and have the staging toggle on, but all the network requests are still going to production

**Comment by mountiny** [comment]
I checked the logs and I can confirm this also happens on iOS 😱 I think we havent really tested this in phone....

@mdneyazahmad Would you be able to look into this as regression. Not sure why mobile would behave differently

**Comment by Santhosh-Sellavel** [comment]
Weird was it always hitting production or randomly between staging/production when staging is enabled?

**Comment by mountiny** [comment]
I just saw staging in my testing, I tried toggling it on and off and it always hit production afaik

**Comment by iwiznia** [comment]
Always production for me.

**Comment by mountiny** [comment]
@mdneyazahmad Will you be able to look into this?

**Comment by mdneyazahmad** [comment]
@mountiny Sorry for the delay. As it does not work on native devices. I guess, it has something to do with environment variable setup. Maybe this variable is false.

https://github.com/Expensify/App/blob/6748e8a6c5363a626490e436c84ecd53df7266c6/src/CONFIG.js#L59

Could you provide me with the steps to see logs on staging app?

**Comment by mountiny** [comment]
> Could you provide me with the steps to see logs on staging app?

That is unfortunately only internal tool.

Wondering if there is away we could debug this without this tool.

**Comment by mdneyazahmad** [comment]
Could we log `IS_IN_STAGING` variable to see it is indeed true and the api is hitting production server?

**Comment by mountiny** [comment]
Looking into this

**Comment by mountiny** [comment]
Alright, chatted with our App deployers and this is a problem out of scope for this one, I created a new issue to discover our options https://github.com/Expensify/App/issues/11561 and you can find context there. Thanks!

**Comment by mountiny** [comment]
Idk I think this was hard to test and catch in the PR as the contributor and C+ could not expect that the native release will not consider the env variables. I would not treat this as regression because of that circumstances and having a new issue to explore solutions to this which might not be trivial seems fair to me (and also clearer than to reopen this one).

**Comment by flodnv** [comment]
It looks like the QA steps of [the PR](https://github.com/Expensify/App/pull/10690) were wrong (should've been verified by an engineer). I suppose it's fine to open a new issue, but I don't think it's fair to say that this was out of scope when it's in the title of the issue.

**Comment by mountiny** [comment]
I agree I should have made sure the iOS/Android is thoroughly tested instead of just assuming it will work if the web/desktop works.

> I don't think it's fair to say that this was out of scope when it's in the title of the issue

What I mean is more like I dont want to throw this back to the contributor or treat it as regression given it was my, C+ and QA mistake to not think of testing this on native platforms. I dont think the contributor should be penalized in any way for my mistake, although we already paid them out.

So in case we want to open the new issue to new contributors then having a specific issue is clearer imo.

**Comment by flodnv** [comment]
Yeah, let's continue in the new issue

**Comment by iwiznia** [comment]
I would say that we should ask the contributor to fix it. Just because we did not catch it, does not mean that the job they were hired for was not really done.
This is similar to when contributors ask for more payment because the issue took longer or was harder than expected. We recognize that and pay them extra. This is kind of the same but the other way around. We can all chose to say: what we agreed is what we agreed and there's no wiggle room OR we can chose to be more flexible and recognize that these situations (issues being more complex than expected and not catching issues during review) will exist. I rather be more flexible and permissive.

**Comment by mountiny** [comment]
> I would say that we should ask the contributor to fix it. Just because we did not catch it, does not mean that the job they were hired for was not really done.

if it was in the regression period, then I agree.

**Comment by iwiznia** [comment]
It was even worse than a regression, it was something that was supposed to be done and was not.

**Comment by mdneyazahmad** [comment]
I am sorry that the issue is not fully fixed here. I agree that the title suggests it should be fixed in this issue, but the proposed solution here assumes that the environment variable is properly set, and the solution is build on top of that. It was not easy to test the url of the request on the staging app, and it got missed. I would like to work on the issue created, and consider that as a regression. Thank you.

**Comment by mountiny** [comment]
@mdneyazahmad Thank you! We learn something new about this App everyday 😂

**Comment by iwiznia** [comment]
Thanks you very much @mdneyazahmad!

**Comment by mountiny** [comment]
Let's just add proposals to the new issue 😅 https://github.com/Expensify/App/issues/11561

This way it wont be blocked only on one contributor and theoretically we can get working solution faster than if we would leave this for @mdneyazahmad only (and they can propose solutions there too)

**Comment by flodnv** [comment]
Agreed, let's move to https://github.com/Expensify/App/issues/11561, and @mdneyazahmad we'd be very happy to hire you for this if you have a good proposal!
